index.js

```diff
@@ -45,21 +56,18 @@ function isCurrentCityDataLoaded() {
   return cityByWeather.hasOwnProperty(currentCity);
 }
 
-renderApp();
-
 function reRenderApp() {
   // Need to split rendering cycles
-  setTimeout(() => window.renderApp(), 0);
+  setTimeout(window.renderApp, 0);
 }
 
-function loadData() {
+function validateAndLoadData() {
```

Comment:
If `validateAndLoadData` was fully thenable (i.e. returned a promise, resolved in some cases), then we could fire re-render beyond it.

This would also enable network errors handling.